Outliers

w-alanna opened the pull request:
Adds the outlier column, shows the plot that is the the outlier, shows N/A if there is no outlier

w-alanna committed:
Added median absolute deviation to dataframe

w-alanna committed:
Added column that shows the plot with the outlier sensor

bpbond commented on qaqc.R:
```diff
@@ -22,10 +20,12 @@ compute_na_sd <- function(filename, mainsite) {
               site = mainsite,
-              plot = substring(design_link,tail(unlist(gregexpr(mainsite, design_link)))+4,nchar(design_link))) %>%
+              plot = substring(design_link,tail(unlist(gregexpr(mainsite, design_link)))+4,nchar(design_link)),
+              outlier = if_else(value > average+MAD | value < average-MAD, plot, "N/A")) %>%
```
1. Where is the MAD computation?
2. Use R's `NA()` instead of string "N/A"

w-alanna replied:
The MAD computation is on line 13 and I changed the "N/A"

w-alanna committed:
update outliers

I changed "N/A" to NA and MAD is mutated into compassData before the summarise in line 13

bpbond commented on qaqc.R:
```diff
@@ -10,9 +10,7 @@ compute_na_sd <- function(filename, mainsite) {
-  
-  #test
-  test <- nrow(compassData$value)
+  compassData <- compassData %>% group_by(research_name, date) %>% mutate(MAD=mad(value, const=1, na.rm=TRUE, low=FALSE, high=FALSE))
```
👍 

Why are you changing the `const` parameter from its default value?

Also, you don't need to say `low=FALSE, high=FALSE` since those are already the default values.

w-alanna replied:
I wasn't sure what the default value was for const so I made it 1

w-alanna committed:
Removed default values from mad()

w-alanna committed:
Removed const=1 so the default value will be used
